src/main/java/io/ticktok/client/tick/http/HttpTickerPolicy.java

```diff
@@ -0,0 +1,95 @@
+package io.ticktok.client.tick.http;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import io.ticktok.client.tick.TickChannel;
+import io.ticktok.client.tick.TickConsumer;
+import io.ticktok.client.tick.TickConsumerInvoker;
+import io.ticktok.client.tick.TickerPolicy;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.DefaultHttpRequestRetryHandler;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.util.Timer;
+import java.util.TimerTask;
+
+@Slf4j
+public class HttpTickerPolicy implements TickerPolicy {
+
+    public static final String URL_PARAM = "url";
+    private final HttpClient httpClient = HttpClients.custom()
```

Comment:
is there a test for retry?

Reply:
No, retry is a tricky state which I didn't managed to simulate
if you have an idea how to test it it would be great

Reply:
i implemented this feature in a different project with kotlin, i think you can take sum ideas from.

**production code:**

retry util:
```
    private inline fun repeatUntil(block:() -> String): String {
        val counter = 3
        for(i in 1..counter) {
            try {
                return block()
            } catch (e: IOException) { // catch all .net error
                logger.warn().message("API call failed, reason: ${e.message}, going to retry call").log()
            }
        }

        throw ClientRequestException("Connection refused, Fail call service after $counter times" )
    }
```

client usage:
```
 fun assemble(): String {
        val response = repeatUntil{ makeReport() }
        validateResponse(response)
        return downloadReport(response)
    }
```

**Test code:**

```
    @Test()
    fun retryMax3TimesOnFail(){
        assertThrows<ClientRequestException> {
            client("/maxRetry").assemble()
        }
        assertThat(failExpensifyStub.maxRetryTimes, Is.`is`(3))
    }

    private fun client(domain:String) =
            ExpensifyClient("http://localhost:$STUB_PORT$domain", "my_username", "my_password", "2018-02-01")

    @Test
    fun retry1TimeOnFail(){
        client("/singleRetry").assemble()
        assertThat(failExpensifyStub.singleRetry, Is.`is`(1))
    }

    object failExpensifyStub {
        var maxRetryTimes = 0
        var singleRetry = 0

        fun init() {
            Javalin.create().enableCaseSensitiveUrls().start(STUB_PORT).routes {
                ApiBuilder.post("singleRetry") { context ->
                    when(singleRetry == 1) {
                        true -> context.status(200)
                        false -> {
                            singleRetry++
                            context.status(500)
                        }
                    }
                }
                ApiBuilder.post("maxRetry") { context ->
                    failExpensifyStub.maxRetryTimes++
                    context.status(500)
                }
            }
        }
    }
```

Reply:
The problem is that you're not conforming to the standard retries 
Standard retries retries on connection lost and not on every IOException 
and it's actually makes sense IMO